src/components/MapView/Analyser/index.tsx

```diff
@@ -0,0 +1,402 @@
+  const analysisResult = useSelector(analysisResultSelector);
+
+  const isAnalysisLoading = useSelector(isAnalysisLoadingSelector);
+  const isMapLayerActive = useSelector(isAnalysisLayerActiveSelector);
+
+  const [isAnalyserFormOpen, setIsAnalyserFormOpen] = useState(false);
+  const [isTableViewOpen, setIsTableViewOpen] = useState(true);
+
+  // form elements
+  const [hazardLayerId, setHazardLayerId] = useState<LayerKey>();
+  const [statistic, setStatistic] = useState(AggregationOperations.Mean);
+  const [baselineLayerId, setBaselineLayerId] = useState<LayerKey>();
+  const [selectedDate, setSelectedDate] = useState<number | null>(null);
+
+  const [belowThreshold, setBelowThreshold] = useState('');
+  const [aboveThreshold, setAboveThreshold] = useState('');
+  const [thresholdError, setThresholdError] = useState<string | null>(null);
+
+  // enforce errors
+  useEffect(() => {
+    const belowThresholdValue = parseFloat(belowThreshold);
+    const aboveThresholdValue = parseFloat(aboveThreshold);
+    if (belowThresholdValue < aboveThresholdValue) {
+      setThresholdError('Min threshold is larger than Max!');
+    } else {
+      setThresholdError(null);
+    }
+  }, [belowThreshold, aboveThreshold]);
+
+  // set default date after dates finish loading and when hazard layer changes
+  useEffect(() => {
+    const dates = hazardLayerId
+      ? availableDates[
+          (LayerDefinitions[hazardLayerId] as WMSLayerProps)?.serverLayerName
+        ]
+      : null;
+    if (!dates || dates.length === 0) {
+      setSelectedDate(null);
+    } else {
+      setSelectedDate(dates[dates.length - 1]);
+    }
+  }, [availableDates, hazardLayerId]);
+
+  const onOptionChange = <T extends string>(
+    setterFunc: Dispatch<SetStateAction<T>>,
+  ) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    setterFunc(event.target.value as T);
+  };
+
+  const adminBoundariesExtent = useMemo(() => {
+    if (!boundaryLayerData) {
+      // not loaded yet. Should be loaded in MapView
+      return null;
+    }
+    return bbox(boundaryLayerData.data) as Extent; // we get extents of admin boundaries to give to the api.
+  }, [boundaryLayerData]);
+
+  const statisticOptions = Object.entries(AggregationOperations).map(stat => (
+    <FormControlLabel
+      key={stat[0]}
+      value={stat[1]}
+      control={
+        <Radio className={classes.radioOptions} color="default" size="small" />
+      }
+      label={stat[0]}
+    />
+  ));
+
+  const runAnalyser = async () => {
+    if (!adminBoundariesExtent) {
+      return;
+    } // hasn't been calculated yet
+    if (analysisResult) {
+      // if one exists we are likely trying to clear it to do a new one
+      dispatch(clearAnalysisResult());
+      return;
+    }
+    if (!selectedDate) {
+      throw new Error('Date must be given to run analysis');
+    }
+
+    if (!hazardLayerId || !baselineLayerId) {
+      throw new Error('Layer should be selected to run analysis');
+    }
```

Comment:
these errors are more so to make TS happy, as TS already ensures non-null values are passed into the thunk. If these are thrown, then it highlights a clear developer-side issue as the fields weren't validated properly before `runAnalysis` was called. (The button is disabled if the fields are invalid/empty). Do you think it should be done a different way?

Reply:
Gotcha, I see the issue. I think in a perfect world we'd handle these with form validations, and make sure that TS knows that this function will never be called without the form validations being run. But as long as we have form validations/user-friendly warnings there, it's probably fine for these to live behind the scenes. If we're relying on these to _actually_ enforce the conditions (rather than just to make TS happy), we should make the errors visible though...